common/flatpak-run.c

```diff
@@ -1622,13 +1622,31 @@ job_removed_cb (SystemdManager *manager,
     g_main_loop_quit (data->main_loop);
 }
 
+static gchar *
+systemd_unit_name_escape (const gchar *in) {
+  /* Adapted from systemd source */
+  gchar * const ret = g_new (gchar, strlen (in) * 4 + 1);
```

Comment:
Can you use GString instead of the custom allocation and append code. g_string_append_printf() is nice for this sort of thing.

Also, the indentation isn't flatpak standard.

Reply:
done